js-api-spec/compiler.test.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {initCompiler, initAsyncCompiler} from 'sass';
+import type {Compiler, AsyncCompiler} from 'sass';
+import {sandbox} from './sandbox';
+
+describe('Compiler', () => {
+  let compiler: Compiler;
+  beforeEach(() => {
+    compiler = initCompiler();
+  });
+  describe('compileString', () => {
+    it('performs multiple compilations', () => {
+      expect(compiler.compileString('$a: b; c {d: $a}').css).toBe(
+        'c {\n  d: b;\n}'
+      );
+      expect(compiler.compileString('$a: 1; c {d: $a}').css).toBe(
+        'c {\n  d: 1;\n}'
+      );
+    });
+    it('throws after being disposed', () => {
+      compiler.dispose();
+      expect(() => compiler.compileString('$a: b; c {d: $a}')).toThrow();
+    });
+  });
+  describe('compile', () => {
+    it('performs multiple compilations', () => {
+      sandbox(dir => {
+        dir.write({'input.scss': '$a: b; c {d: $a}'});
+        dir.write({'input2.scss': '$a: 1; c {d: $a}'});
+        expect(compiler.compile(dir('input.scss')).css).toBe('c {\n  d: b;\n}');
+        expect(compiler.compile(dir('input2.scss')).css).toBe(
+          'c {\n  d: 1;\n}'
+        );
+      });
+    });
+    it('throws after being disposed', () => {
+      sandbox(dir => {
+        dir.write({'input.scss': '$a: b; c {d: $a}'});
+        compiler.dispose();
+        expect(() => compiler.compile(dir('input.scss'))).toThrow();
+      });
+    });
+  });
+});
+
+describe('AsyncCompiler', () => {
+  let compiler: AsyncCompiler;
+  beforeEach(async () => {
+    compiler = await initAsyncCompiler();
+  });
+  describe('compileStringAsync', () => {
+    it('performs multiple compilations', async () => {
+      expect((await compiler.compileStringAsync('$a: b; c {d: $a}')).css).toBe(
+        'c {\n  d: b;\n}'
+      );
+      expect((await compiler.compileStringAsync('$a: 1; c {d: $a}')).css).toBe(
+        'c {\n  d: 1;\n}'
+      );
+    });
+    it('throws after being disposed', async () => {
+      await compiler.dispose();
+      await expectAsync(
+        compiler.compileStringAsync('$a: b; c {d: $a}')
+      ).toThrowSassException({line: 0});
```

Comment:
Even though I'm expecting the throw here, the tests are still failing with:

```
AsyncCompiler compileStringAsync throws after being disposed
  - Error: Compiler caused error: Async compiler has already been disposed..
```

Am I doing something wrong here? https://github.com/oddbird/embedded-host-node/blob/ebfddd959e77c98a0a8ee347a40efb24ee2a24e2/lib/src/async-compiler.ts#L62-L66

Reply:
I switched to a regular `expect().toThrow()` in https://github.com/sass/sass-spec/pull/1954/commits/6145b4e73263402227b91140c94d9067b257906f. That makes the test pass, but I'm curious if that's the correct behavior

Reply:
> I switched to a regular `expect().toThrow()` in [6145b4e](https://github.com/sass/sass-spec/commit/6145b4e73263402227b91140c94d9067b257906f). That makes the test pass, but I'm curious if that's the correct behavior

@jerivas Looking at similar tests in `compile.test.ts`, yes I think this is the correct behavior. I updated them to use `toThrowError`, but that's minor.